### PR TITLE
feat: [SRM-10747]: As a Harness CI/NG customer, I want to be able tur…

### PR DESCRIPTION
**Pull request by angelo-m-rodriguez**
…n on Error Tracking in the Run Tests step in the Build Stage

##### Jira Links:

https://harness.atlassian.net/browse/SRM-10747

##### Screenshots:

Screen Shot 2022-06-23 at 2 47 19 PM

Screen Shot 2022-06-23 at 2 46 22 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Review by DFieldFL — CHANGES_REQUESTED** [review]

**Review by DFieldFL — DISMISSED** [review]

File: src/modules/75-ci/components/PipelineSteps/RunTestsStep/RunTestsStepBase.tsx [hunk 1]
```diff
@@ -451,6 +482,53 @@ export const RunTestsStepBase = (
                 allowableTypes: [MultiTypeInputType.FIXED]
               })}
             </Container>
+            {ERROR_TRACKING_ENABLED && (formik.values?.spec?.language as any)?.value === Language.Java && (
```
Comment (mtran7) [inline src/modules/75-ci/components/PipelineSteps/RunTestsStep/RunTestsStepBase.tsx:485]: can use `selectedLanguageValue`

File: src/modules/75-ci/components/PipelineSteps/RunTestsStep/RunTestsStepBase.tsx [hunk 2]
```diff
@@ -451,6 +482,53 @@ export const RunTestsStepBase = (
...
+              <>
+                <Text font={{ size: 'small' }}>{getString('ci.runTestsErrorTrackingSetupText')}</Text>
+                <RadioButtonGroup
+                  name="error-tracking-setup"
+                  inline={true}
+                  selectedValue={isErrorTrackingCurrentlyOn ? 'yes' : 'no'}
```
Comment (mtran7) [inline src/modules/75-ci/components/PipelineSteps/RunTestsStep/RunTestsStepBase.tsx:491]: can you create an enum for yes/no. 

Ex. isErrorTrackingCurrentlyOn ? ErrorTrackingOn.YES : ErrorTrackingOn.No

File: src/modules/75-ci/components/PipelineSteps/RunTestsStep/RunTestsStepBase.tsx [hunk 3]
```diff
@@ -451,6 +482,53 @@ export const RunTestsStepBase = (
                 allowableTypes: [MultiTypeInputType.FIXED]
               })}
             </Container>
+            {ERROR_TRACKING_ENABLED && (formik.values?.spec?.language as any)?.value === Language.Java && (
+              <>
+                <Text font={{ size: 'small' }}>{getString('ci.runTestsErrorTrackingSetupText')}</Text>
```
Comment (mtran7) [inline src/modules/75-ci/components/PipelineSteps/RunTestsStep/RunTestsStepBase.tsx:487]: Add a ui anchor here so Docs can provide a tooltip on what "Error Tracking" is.

https://harness.atlassian.net/wiki/spaces/CDNG/pages/1626144816/NG+Tooltip+Framework+-+self+help+guide+for+docs

**Review by DFieldFL — DISMISSED** [review]

File: src/modules/75-ci/components/PipelineSteps/RunTestsStep/RunTestsStepBase.tsx (hunk 1 above)
Comment (angelo-m-rodriguez) [reply src/modules/75-ci/components/PipelineSteps/RunTestsStep/RunTestsStepBase.tsx]: Made use of selectedLanguageValue

File: src/modules/75-ci/components/PipelineSteps/RunTestsStep/RunTestsStepBase.tsx [hunk 4]
```diff
@@ -405,6 +430,12 @@ export const RunTestsStepBase = (
         setFormikRef?.(formikRef, formik)
         const selectedLanguageValue = (formik.values?.spec?.language as any)?.value
         const buildTool = (formik.values?.spec?.buildTool as any)?.value
+        const isErrorTrackingCurrentlyOn =
```
[the remark this answers is not in the document]
Comment (angelo-m-rodriguez) [reply src/modules/75-ci/components/PipelineSteps/RunTestsStep/RunTestsStepBase.tsx:466]: Move from onChange into method called getUpdatedPreCommand and referenced method inside onChange

File: src/modules/75-ci/components/PipelineSteps/RunTestsStep/RunTestsStepBase.tsx (hunk 3 above)
Comment (angelo-m-rodriguez) [reply src/modules/75-ci/components/PipelineSteps/RunTestsStep/RunTestsStepBase.tsx]: Added dataTooltipId to tooltipProps

File: src/modules/75-ci/components/PipelineSteps/RunTestsStep/RunTestsStepBase.tsx [hunk 5]
```diff
@@ -451,6 +482,53 @@ export const RunTestsStepBase = (
...
+            {ERROR_TRACKING_ENABLED && (formik.values?.spec?.language as any)?.value === Language.Java && (
+              <>
+                <Text font={{ size: 'small' }}>{getString('ci.runTestsErrorTrackingSetupText')}</Text>
+                <RadioButtonGroup
+                  name="error-tracking-setup"
+                  inline={true}
+                  selectedValue={isErrorTrackingCurrentlyOn ? 'yes' : 'no'}
+                  onChange={(e: FormEvent<HTMLInputElement>) => {
+                    const preCommand = formik?.values?.spec?.preCommand as string
+                    const turnErrorTrackingOn = e.currentTarget.value === 'yes'
```
[the remark this answers is not in the document]
Comment (angelo-m-rodriguez) [reply src/modules/75-ci/components/PipelineSteps/RunTestsStep/RunTestsStepBase.tsx]: Replaced reference to hardcoded strings with reference to newly created ErrorTrackingStatus enum

**Review by DFieldFL — APPROVED** [review]
🥇

**Comment by angelo-m-rodriguez** [comment]
retrigger titlecheck

**Review by DFieldFL — APPROVED** [review]

**Comment by DFieldFL** [comment]
trigger cypress

**Comment by DFieldFL** [comment]
trigger cypress

**Comment by DFieldFL** [comment]
trigger cypress

**Comment by DFieldFL** [comment]
retrigger jest

**Pull request merged**